tests/unix_stream.rs

```diff
@@ -445,6 +457,8 @@ where
 
     assert!(stream.take_error().unwrap().is_none());
 
+    assert_would_block(stream.read(&mut buf));
```

Comment:
Why is this needed? Ideally the test are unchanged as the behaviour is already dependent on in existing code.

Reply:
We need to make a call that would block to reset the interests before trying to write per https://docs.rs/mio/latest/mio/struct.Poll.html#draining-readiness. I believe right now this test does not fully follow this documentation.